accounts/views.py

```diff
@@ -25,11 +28,35 @@ def form_valid(self, form):
             login(self.request,user)
             return response
         
+class Login(LoginView):#ログインページ
+    form_class = LoginForm
+    template_name = 'accounts/login.html'
+
+class Logout(LoginRequiredMixin,LogoutView):#ログアウトページ
+    template_name = 'accounts/login.html'
+
+class UserProfileView(LoginRequiredMixin,DetailView):
+    model = Profile
+    template_name = 'accounts/profile.html'
+
+class UserProfileEditView(LoginRequiredMixin,UserPassesTestMixin,UpdateView):
+    model = Profile
+    form_class = ProfileForm
+    template_name = 'accounts/profile_edit.html'
+
+    def get_success_url(self):
+        return reverse_lazy('accounts:user_profile',kwargs={'pk':self.object.pk})#こっちだと成功 おそらくform_class=を提示しているから

```

Comment:
```suggestion
        return reverse('accounts:user_profile', kwargs={'pk':self.object.pk})

```
で大丈夫です。[参考](https://btj0.com/blog/django/success_url-get_success_url-reverse-reverse_lazy/)
また、get_success_urlメソッドをオーバーライドしないと成功しない理由は、form_classを提示しているためではなく、kwargs={'pk':self.object.pk}があるためです。このように、pkをurlに含む場合にアトリビュートにsuccess_urlを書いたら失敗するのは、どのオブジェクトを参照すればいいか不明なためです。そこで、メソッドのself引数からpkの参照オブジェクトを特定させています。

Reply:
アドバイスありがとうございます。訂正しました